tests/core/integrations/test_diarization.py

```
@@ -0,0 +1,540 @@
+    """diarize() raises DiarizationError with license link on HTTP 403."""
+    result = _sample_result()
+    mock_pipeline_cls = MagicMock()
+    load_exc = Exception("forbidden")
+    load_exc.status_code = 403  # type: ignore[attr-defined]
+    mock_pipeline_cls.from_pretrained.side_effect = load_exc
+
+    with patch(
+        "insanely_fast_whisper_rocm.core.integrations.diarization.Pipeline",
+        mock_pipeline_cls,
+    ):
+        with pytest.raises(DiarizationError, match="huggingface.co"):
```

Ruff (0.15.11)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)